File: 180501/codefight/arcade/JungDongHyun/43_isBeautifulString.py
def isBeautifulString(input_string):
    offset = ord('a')
    bucket = [0 for i in range(26)]
    for ch in input_string:
        bucket_idx = ord(ch) - offset
        bucket[bucket_idx] += 1

    prev_occ = bucket[0]
    is_beautiful = True
    for occ in bucket:
        if prev_occ < occ:
            is_beautiful = False
            break
        prev_occ = occ

    return is_beautiful

File: 180501/codefight/arcade/NaDongHyun/43_isBeautifulString.js
function isBeautifulString(inputString) {
  var a = inputString.split('');
  var bucket = [];
  for(var i = 0; i < 26; i ++) {
      bucket.push(0);
  }
  for(var i = 0; i < inputString.length; i++) {
      bucket[a[i].charCodeAt() - 97] += 1;
  }
  for(var i = 0; i < bucket.length - 1; i ++) {
      if(bucket[i] < bucket[i+1]) {
          return false;
      }
  }
  return true;
}
